homeassistant/components/tts/__init__.py

```diff
@@ -64,7 +64,9 @@
 SERVICE_CLEAR_CACHE = "clear_cache"
 SERVICE_SAY = "say"
 
-_RE_VOICE_FILE = re.compile(r"([a-f0-9]{40})_([^_]+)_([^_]+)_([a-z_]+)\.[a-z0-9]{3,4}")
+_RE_VOICE_FILE = re.compile(
+    r"([a-f0-9]{40})_([^_]+)_([^_]+)_([a-z0-9-_]+)\.[a-z0-9]{3,4}"
+)
```

Comment:
> underscores (en_us)

This reflect not what you change, so I did not understand you change on that

Reply:
You are right, this variant is not robust. I have adapted it to work as intended. I have tested it to work with two different languages and two different TTS providers

Reply:
If you make changes to a core integration, you need to accompany it with tests.

Reply:
I have included a test, that checks whether the regex works on some common filenames.
Please, let me know, if I need to change anything else.